milavision/milavision/envs/mila.py

```diff
@@ -0,0 +1,139 @@
+""" Set of functions for creating torchvision datasets when on the Mila cluster.
+
+IDEA: later on, we could also add some functions for loading torchvision models from a cached
+directory. 
+"""
+import inspect
+import os
+import shutil
+import socket
+from logging import getLogger as get_logger
+from pathlib import Path
+from typing import Dict, List, Optional, Type, Union
+
+import torchvision.datasets as tvd
+from torchvision.datasets import VisionDataset
+
+from milavision._utils import VD, ClusterType
+
+current_env = ClusterType.current()
+# NOTE: It should always be possible to import this module even when not on the mila cluster.
+
+
+""" a map of the files for each dataset type, relative to the torchvision dir. """
+dataset_files: Dict[Type[VisionDataset], List[str]] = {
+    tvd.MNIST: ["MNIST"],
+    tvd.CIFAR10: ["cifar-10-batches-py"],
+    tvd.CIFAR100: ["cifar-100-python"],
+    tvd.ImageNet: ["train", "val"],
+}
+
+logger = get_logger(__name__)
+
+
+def on_login_node() -> bool:
+    # IDEA: Detect if we're on a login node somehow.
+    return socket.getfqdn().endswith(".server.mila.quebec") and "SLURM_TMPDIR" not in os.environ
+
+
+""" a map of the files for each dataset type, relative to `current_env.torchvision_dir`. """
+dataset_files_paths: Dict[Type[VisionDataset], List[Path]] = {
+    k: list(map(Path, v)) for k, v in dataset_files.items()
+}
+
+
+_IGNORED = ""
+
+
+def make_dataset(dataset_type: Type[VD], root: str, *agrs, **kwargs) -> VD:
```

Comment:
```suggestion
def make_dataset(dataset_type: Type[VD], root: str, *args, **kwargs) -> VD:
```

(ditto for uses of the variable below)